x/leverage/keeper/iter.go

```diff
@@ -45,14 +47,22 @@ func (k Keeper) GetAllRegisteredTokens(ctx sdk.Context) []types.Token {
 // GetAllSpecialAssetPairs returns all the special asset pairs from the x/leverage
 // module's KVStore.
 func (k Keeper) GetAllSpecialAssetPairs(ctx sdk.Context) []types.SpecialAssetPair {
-	return store.MustLoadAll[*types.SpecialAssetPair](ctx.KVStore(k.storeKey), types.KeyPrefixSpecialAssetPair)
+	pairs := store.MustLoadAll[*types.SpecialAssetPair](ctx.KVStore(k.storeKey), types.KeyPrefixSpecialAssetPair)
+	sort.SliceStable(pairs, func(i, j int) bool {
+		return pairs[i].CollateralWeight.LT(pairs[j].CollateralWeight)
+	})
+	return pairs
```

Comment:
We don't need sorting for grpc_query ,
```suggestion
	func GetAllSpecialAssetPairs(ctx sdk.Context, sort bool){
	prefix := types.KeySpecialAssetPairOneDenom(denom)
	pairs := store.MustLoadAll[*types.SpecialAssetPair](ctx.KVStore(k.storeKey), prefix)
	if sort {
		sort.SliceStable(pairs, func(i, j int) bool {
			return pairs[i].CollateralWeight.LT(pairs[j].CollateralWeight)
		})
	}
	return pairs
}
```